src/components/DependencyGraph.tsx

```diff
@@ -41,13 +41,19 @@ export const DependencyGraph: React.FC = () => {
 
     return (
         <div className={containerCls}>
-            <Radio
-                onChange={(e: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => setShowOnlyConnectedNodes(data.checked)}
-                toggle
-                checked={showOnlyConnectedNodes}
-            />
+            {process.env.NODE_ENV === 'development' && (
+                <>
+                    <Radio
+                        onChange={(e: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => setShowOnlyConnectedNodes(data.checked)}
```

Comment:
Instead of creating an anonymous function on each re-render, maybe you can utilize the useCallback hook?

Reply:
Fixed